os/kernel/sched/sched_cpuload.c

```diff
@@ -273,4 +306,49 @@ int clock_cpuload(int pid, int index, FAR struct cpuload_s *cpuload)
 	irqrestore(flags);
 	return ret;
 }
+
+/****************************************************************************
+ * Function:  clock_cpuload_snapshot
+ *
+ * Description:
+ *   Return load measurement data for the select PID in specific time interval.
+ *
+ * Parameters:
+ *   pid - The task ID of the thread of interest.  pid == 0 is the IDLE thread.
+ *   cpuload - The location to return the CPU load
+ *
+ * Return Value:
+ *   OK (0) on success; a negated errno value on failure.  The only reason
+ *   that this function can fail is if 'pid' no longer refers to a valid
+ *   thread.
+ *
+ * Assumptions:
+ *
+ ****************************************************************************/
+
+int clock_cpuload_snapshot(int pid, FAR struct cpuload_s *cpuload)
+{
+	int tick_index;
+	int hash_index;
+
+	hash_index = PIDHASH(pid);
+	DEBUGASSERT(cpuload);
+
+	/* Check if the entry is valid (TCB field is not NULL) */
+	if (g_pidhash[hash_index].tcb == NULL) {
+		return -ESRCH;
+	}
+
+	cpuload->active = 0;
+	cpuload->total = g_cpusnap_ticks;
```

Comment:
When arr is not full-filled, is it ok to use the size of arr?

Reply:
yes, cpuload monitor is started after snapshot interval (g_cpusnap_ticks) when snapshot mode.